planex/extract.py

```diff
@@ -98,7 +98,7 @@ def archive_root(tar):
     """
     Return the name of the top level directory of the tarball
     """
-    if tar.firstmember.type == tarfile.DIRTYPE:
+    if len(tar.members) == 1 and tar.members[0].isdir():
```

Comment:
I'm surprised about `len(tar.members) == 1`. Does this mean the tar can only have one entry? Other tars still could have a top-level directory that this function would not return.